tests/test_gen2.py

```diff
@@ -41,33 +41,39 @@ def createConfig(self, config_name, fits_name, mapper):
             config = yaml.safe_load(f)
 
         ingesterConfig = config["ingester"]
-        dataDir = tempfile.mkdtemp()
-        ingesterConfig["forwarderStagingDirectory"] = dataDir
+        self.forwarderStagingDirectory = tempfile.mkdtemp()
+        ingesterConfig["forwarderStagingDirectory"] = self.forwarderStagingDirectory
 
-        badDir = tempfile.mkdtemp()
         butlerConfig = config["ingester"]["butlers"][0]["butler"]
-        butlerConfig["badFileDirectory"] = badDir
 
-        butlerStageDir = tempfile.mkdtemp()
-        butlerConfig["stagingDirectory"] = butlerStageDir
+        self.badDir = tempfile.mkdtemp()
+        butlerConfig["badFileDirectory"] = self.badDir
 
-        repoDir = tempfile.mkdtemp()
-        butlerConfig["repoDirectory"] = repoDir
+        self.stagingRootDir = tempfile.mkdtemp()
+        butlerConfig["stagingDirectory"] = self.stagingRootDir
 
-        destFile = os.path.join(dataDir, fits_name)
-        copyfile(fitsFile, destFile)
+        self.repoDir = tempfile.mkdtemp()
+        butlerConfig["repoDirectory"] = self.repoDir
 
-        mapperFileName = os.path.join(repoDir, "_mapper")
+        subDir = tempfile.mkdtemp(dir=self.forwarderStagingDirectory)
+        self.destFile = os.path.join(subDir, fits_name)
+        copyfile(fitsFile, self.destFile)
+
+        mapperFileName = os.path.join(self.repoDir, "_mapper")
         with open(mapperFileName, 'w') as mapper_file:
             mapper_file.write(mapper)
 
-        return config, destFile, repoDir, badDir
+        return config
+
+    def strip_prefix(self, name, prefix):
```

Comment:
Maybe you want to fix these two test functions as well?

Reply:
I fixed those too, just now.